File: sync/src/main/scala/com/fan/mysql/parser/FilterableParser.java
package com.dingcloud.dts.binlog.parser;

import com.dingcloud.dts.binlog.BinlogEventFilter;
import com.dingcloud.dts.binlog.BinlogEventParser;

public abstract class FilterableParser implements BinlogEventParser {

	protected BinlogEventFilter filter;

	public void setFilter(BinlogEventFilter filter) {
		this.filter = filter;
	}
}

File: async/src/main/scala/com/fan/mysql/async/binlog/parse/FilterableParser.scala
package com.fan.mysql.async.binlog.parse

import com.fan.mysql.async.binlog.{BinlogEventFilter, BinlogEventParser}

abstract class FilterableParser extends BinlogEventParser {

  protected var filter: BinlogEventFilter = _

  def setFilter(filter: BinlogEventFilter): Unit = {
    this.filter = filter
  }
}
